mypyc/test-data/irbuild-tuple.test

```diff
@@ -874,3 +874,226 @@ L2:
     r3 = 1
 L3:
     return r3
+
+[case testTupleFromRange1]
+def fn() -> tuple[int, ...]:
+    return tuple(range(3))
```

Comment:
Similar to above, `tuple(range(...))` seems very rare (based on a large codebase I have access to -- there were no instance of it). Again, having run tests seems sufficient since irbuild tests are verbose and take some maintenance, preferable added to an existing run test.

If you create a PR that optimizes this, I wouldn't spend extra effort on `tuple` specifically, but if an optimization also benefits tuples, it would be enough to have some example IR dumps for relevant code in the PR summary.